modules/core/src/main/java/org/apache/ignite/cache/query/VectorQuery.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cache.query;
+
+import javax.cache.Cache;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.query.annotations.QueryVectorField;
+import org.apache.ignite.internal.processors.query.QueryUtils;
+import org.apache.ignite.internal.util.typedef.internal.A;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ * Vector queries based on Apache Lucene engine.
+ * <h1 class="header">Description</h1>
+ * Ignite supports vector queries based on Apache Lucene engine.
+ * Note that all fields that are expected to show up in vector query results must be annotated with {@link QueryVectorField}
+ * and be of type {@code float[]}.
+ *
+ * <h2 class="header">Query usage</h2>
+ * As an example, suppose we have data model consisting of {@code 'Employee'} class defined as follows:
+ * <pre name="code" class="java">
+ * public class Person {
+ *     private long id;
+ *
+ *     private String name;
+ *
+ *     // Index for text search.
+ *     &#64;QueryVectorField
+ *     private float[] resume;
+ *     ...
+ * }
+ * </pre>
+ *
+ * Here is a possible query that will use Lucene vector search to scan all resumes to
+ * check if employees have {@code Master} degree:
+ * <pre name="code" class="java">
+ * Query&lt;Cache.Entry&lt;Long, Person&gt;&gt; qry =
+ *     new Vector(Person.class, "resume", toEmbedding("Master"));
```

Comment:
It looks like the `k` parameter is missing

Reply:
good catch, fixed.